pac4j-couch/src/main/java/org/pac4j/couch/profile/service/CouchProfileService.java

```diff
@@ -0,0 +1,181 @@
+package org.pac4j.couch.profile.service;
+
+import org.pac4j.couch.profile.CouchProfile;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.ektorp.CouchDbConnector;
+import org.ektorp.DocumentNotFoundException;
+import org.ektorp.ViewQuery;
+import org.ektorp.ViewResult;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.credentials.password.PasswordEncoder;
+import org.pac4j.core.profile.definition.CommonProfileDefinition;
+import org.pac4j.core.profile.service.AbstractProfileService;
+import org.pac4j.core.util.CommonHelper;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The MongoDB profile service.
+ *
+ * @author Elie Roux
+ * @since 2.0.0
+ */
+public class CouchProfileService extends AbstractProfileService<CouchProfile> {
+
+    private CouchDbConnector couchDbConnector;
+    
+    public static final String COUCH_ID = "_id";
+    public static ObjectMapper objectMapper = new ObjectMapper();
+
+    public CouchProfileService() {
+    	setIdAttribute(COUCH_ID);
+    }
+
+    public CouchProfileService(final CouchDbConnector couchDbConnector) {
+    	setIdAttribute(COUCH_ID);
+        this.couchDbConnector = couchDbConnector;
+    }
+
+    public CouchProfileService(final CouchDbConnector couchDbConnector, final String attributes) {
+    	setIdAttribute(COUCH_ID);
+        this.couchDbConnector = couchDbConnector;
+        setAttributes(attributes);
+    }
+
+    public CouchProfileService(final CouchDbConnector couchDbConnector, final String attributes, final PasswordEncoder passwordEncoder) {
+    	setIdAttribute(COUCH_ID);
+        this.couchDbConnector = couchDbConnector;
+        setAttributes(attributes);
+        setPasswordEncoder(passwordEncoder);
+    }
+
+    public CouchProfileService(final CouchDbConnector couchDbConnector, final PasswordEncoder passwordEncoder) {
+    	setIdAttribute(COUCH_ID);
+        this.couchDbConnector = couchDbConnector;
+        setPasswordEncoder(passwordEncoder);
+    }
+
+    @Override
+    protected void internalInit(final WebContext context) {
+        CommonHelper.assertNotNull("passwordEncoder", getPasswordEncoder());
+        CommonHelper.assertNotNull("couchDbConnector", this.couchDbConnector);
+        defaultProfileDefinition(new CommonProfileDefinition<>(x -> new CouchProfile()));
+
+        super.internalInit(context);
+    }
+
+    @Override
+    protected void insert(final Map<String, Object> attributes) {
+    	logger.debug("Insert doc: {}", attributes);
+    	couchDbConnector.create(attributes);
+    }
+
+    @Override
+    protected void update(final Map<String, Object> attributes) {
+    	try {
+    		final String id = (String) attributes.get(COUCH_ID);
+			final InputStream oldDocStream = couchDbConnector.getAsStream(id);
```

Comment:
there is something wrong with indentation here :)

see what is the standard in other code from pac4j and follow this...
@leleuj we should really add an .editorconfig file in the pac4j repository (like I did in the jax-rs-pac4j and dropwizard-pac4j ones normally).